### ISSUE TITLE
Switch to XML native support library

### ISSUE THREAD

**Issue by PMuellerELCA** [smart-data-lake/smart-data-lake#894]
According to the following [DataBricks Article](https://kb.databricks.com/en_US/data-sources/multiple_xml_data_source-error-while-working-with-xml-data) DataBricks now natively supports XML. So there is no need to load an external library for it (spark_xml).

This additional spark-xml-library also has a special behavior when it comes to XSD validation. It requires that the XSD file is previously uploaded to the spark context with a line like this:

`sc.addFile("dbfs:/demos/xml/invoice.xsd")`
Then, to use this XSD, the xml-option must be `.option("rowValidationXSDPath", "invoice.xsd")`. Without the path.

If you use the new library `org.apache.spark.sql.execution.datasources.xml` the XSD-path-option accepts a regular DBFS path without previous preparations.
`df = spark.read.option("rowTag", "someRow").option("rowValidationXSDPath", "dbfs:/demos/xml/invoice.xsd").option("mode", "PERMISSIVE").format("org.apache.spark.sql.execution.datasources.xml").load("/demos/data/xml/")`

I propose we remove the additional xml library from the pom and redirect the `XmlFileDataObject` to the native library `org.apache.spark.sql.execution.datasources.xml`.

**Comment by pgruetter** [comment]
@PMuellerELCA Remember that Databricks is only one deployment option for SDLB. We can not remove libraries that are available in Databricks, but then break execution locally for example.

**Comment by zzeekk** [comment]
SDLB currently includes the following spark-xml package: https://github.com/databricks/spark-xml

It seems that XML Data Source is integrated into Spark 4.0: https://issues.apache.org/jira/browse/SPARK-44751

Till then we could exclude spark-xml in project pom.xml for projects running on Databricks.

**Comment by iliassoto** [comment]
@PMuellerELCA We can work on this issue as soon as https://github.com/smart-data-lake/smart-data-lake/issues/974 is merged

**Comment by zzeekk** [comment]
Yes, Spark 4 includes an XML data source: https://spark.apache.org/docs/4.0.0/sql-data-sources-xml.html
Scala syntax is `spark.read.xml(...)`, but probably also `spark.read.format('xml').load(...)` works.

**Comment by zzeekk** [comment]
The switch had in fact already happened implicitly: there is no `spark-xml` dependency in the poms anymore, and `format = "com.databricks.spark.xml"` was only resolving through Spark 4's `DataSource.backwardCompatibilityMap` to the built-in `org.apache.spark.sql.execution.datasources.xml`. So the remaining work was to stop relying on that deprecated alias and to drop the workarounds it needed:

- `format` is now `"xml"`.
- Removed the `part-*` → `.xml` renaming: the built-in source already writes `part-<...>.xml` (`XmlFileFormat.getFileExtension`), so the glob never matched anything.
- Removed `assert(partitions.isEmpty, ...)` — `partitionBy` works with the built-in source, so partitioned XML can now be written by SDLB itself. Added a test for write-with-partitions + read back.
- Documented the `rowValidationXSDPath` option: `ValidatorUtil.openSchemaFile` resolves it through the Hadoop filesystem first and only falls back to `SparkFiles.get`, which confirms the observation in this issue — a full path such as `dbfs:/demos/xml/invoice.xsd` works without `SparkContext.addFile` beforehand.

Note that the `xsdfile` schema provider is unrelated to the spark-xml dependency: it uses `org.apache.spark.sql.xml.XsdSchemaConverter` from our own spark-extensions library (derived from spark-xml's XSD support), so that stays.

`XmlFileDataObjectTest` (14 tests) and `CopyActionTest` (14 tests) pass. XML is still a DataSource V1 in Spark 4.1.1 (no V2 XML source in the jar), so the read path in `SparkFileDataObject.getContentV1` is unchanged.
